packages/ui/src/components/ProjectCard/ProjectCard.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Box, Flex, forwardRef, Grid } from '@chakra-ui/react';
+
+import { statusValueCheck } from '../../helpers/statusValueCheck';
+import { Button } from '../Button/Button';
+import { Typography } from '../Typography/Typography';
+
+type DoneProps = {
+  status: 'done';
+  points: number;
+  pointsMax: number;
+  date?: never;
+};
+
+type NotDoneProps = {
+  status: 'idle' | 'doing' | 'review';
+  date: Date;
+  points?: never;
+  pointsMax?: never;
```

Comment:
🤔 Instead of defining prop with type `never` you can simply don't add this prop at all.

Reply:
Adding type never here makes the type safer. This means the component will accept either props date or props points and pointsMax. If we do not add the type never here, Typescript will ask for the appropriate props, but will also accept the extra ones. It will not show an error if I give the status idle, I will give props date and props points.
<img width="968" alt="Zrzut ekranu 2021-07-25 o 16 03 59" src="https://user-images.githubusercontent.com/38866668/126902013-26f5038b-eb39-4a8a-bef3-909257db8f9a.png">